src/som/vmobjects/SDouble.java

```diff
@@ -134,4 +139,92 @@ public SObject primLessThan(final SNumber right, final Universe universe) {
     double r = coerceToDouble(right, universe);
     return asSBoolean(embeddedDouble < r, universe);
   }
+
+  /**
+   * INTEROP
+   * Return Long only
+   */
+
+  @ExportMessage
+  boolean isNumber() {
+    return true;
+  }
+
+  @ExportMessage
+  boolean fitsInByte() {
+    return false;
```

Comment:
Do all of these have to be implemented?
Or would the double one suffice?

Reply:
Hmm, from my understanding (and from the compiler complaints), I must implement all of those as they turn abstract if specify that `SDouble` should be considered as a Number. 
For instance, this is the code for `fitsInByte` in the `InteropLibrary` class:

```
    @Abstract(ifExported = "isNumber")
    public boolean fitsInByte(Object receiver) {
        return false;
    }
```

The library does not offer to chose a type more specific than Number.

Reply:
Hm, sad. Though, could you move the common parts into `SNumber` perhaps?

Reply:
Yes, I was unsure how the annotations would work with subclasses but it seems to work fine. I'll modify SNumber, SDouble and SInteger accordingly. Thanks!